packages/react-components/src/components/AvatarPersona.tsx

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IPersonaProps, Persona, PersonaInitialsColor } from '@fluentui/react';
+import React, { useEffect } from 'react';
+
+/**
+ * Custom data attributes for the AvatarPersona component.
+ */
+export type AvatarPersonaData = {
+  /**
+   * Primary text to display, usually the name of the person.
+   */
+  text?: string;
+  /**
+   * Image URL to use, should be a square aspect ratio and big enough to fit in the image area.
+   */
+  imageUrl?: string;
+  /**
+   * The user's initials to display in the image area when there is no image.
+   * @defaultvalue Derived from `text`
+   */
+  imageInitials?: string;
+  /**
+   * The background color when the user's initials are displayed.
+   * @defaultvalue Derived from `text`
+   */
+  initialsColor?: PersonaInitialsColor | string;
+  /**
+   * The text color when the user's initials are displayed
+   */
+  initialsTextColor?: string;
+};
+
+/**
+ * Callback function used to provide custom data to the AvatarPersona component.
+ */
+export type AvatarPersonaDataProvider = ((userId: string) => Promise<AvatarPersonaData>) | undefined;
+
+export interface AvatarPersonaProps extends IPersonaProps {
+  /**
```

Comment:
What is the benefit of accepting two props `userId` and `dataProvider` as opposed to a single callback that gives us the data we need?

    export interface AvatarPersonaProps extends IPersonaProps {
      getPersonaData: () => Promise<AvatarPersonaData>
    }

Reply:
I also don't quite understand why this is a callback at the component level? Do we expect to use the same `AvatarPersona` component to display personas for different users? Do we expect any of the other persona data to change?

If the intent is simply to support asynchronous loading of persona data, that can be done by the caller by updating the props being passed into the composite.

I understand that we absolutely need a `(userId: string) => PersonaData` callback at higher levels -- e.g. `MessageThread`, `VideoGallery` and the composites, because there are multiple users and we don't a-priori know what users there are.

Reply:
How would the Component pass a `userId` to the callback function `dataProvider` if it doesn't get the `userID` as a prop from composite?

Reply:
This callback is no different than an `onRender` function we have exposed in numerous components.
Instead of returning a custom components, this callback just returns custom data...

Reply:
My liking behind removing the callback is that the callback is async and makes the component no longer pure and dumb - there is an initial render, then a re-render when the async fetch is complete - meaning UI doesn't map 1:1 with props.

You could make it synchronous but then there's no value in it being a callback and may as well just be the props directly.

(disclaimer: could be wrong about this) My thinking on why `onRender` methods are callbacks and not direct props is that a react functional component is a callback so keeping that function signature presumably allows developers to do `onRenderAvatar={<MyAvatar />}`